nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/cosmos/document/AbstractAzureCosmosDBProcessor.java

```diff
@@ -0,0 +1,337 @@
+
+import com.azure.cosmos.ConnectionPolicy;
+import com.azure.cosmos.ConsistencyLevel;
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.CosmosClientException;
+import com.azure.cosmos.CosmosContainer;
+import com.azure.cosmos.CosmosDatabase;
+import com.azure.cosmos.models.CosmosContainerProperties;
+
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.annotation.lifecycle.OnStopped;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.ValidationContext;
+import org.apache.nifi.components.ValidationResult;
+import org.apache.nifi.logging.ComponentLog;
+
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.services.azure.cosmos.AzureCosmosDBConnectionService;
+
+public abstract class AbstractAzureCosmosDBProcessor extends AbstractProcessor {
+    static final String CONSISTENCY_STRONG = "STRONG";
+    static final String CONSISTENCY_BOUNDED_STALENESS= "BOUNDED_STALENESS";
+    static final String CONSISTENCY_SESSION = "SESSION";
+    static final String CONSISTENCY_CONSISTENT_PREFIX = "CONSISTENT_PREFIX";
+    static final String CONSISTENCY_EVENTUAL = "EVENTUAL";
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+    .description("All FlowFiles that are written to Azure Cosmos DB are routed to this relationship").build();
+
+    static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+    .description("All FlowFiles that cannot be written to Azure Cosmos DB are routed to this relationship").build();
+
+    static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("All input FlowFiles that are part of a successful query execution go here.")
+            .build();
+
+    static final PropertyDescriptor CONNECTION_SERVICE = new PropertyDescriptor.Builder()
+        .name("azure-cosmos-db-connection-service")
+        .displayName("Azure Cosmos DB Connection Service")
+        .description("If configured, this property will use the assigned for retrieving connection string info.")
+        .required(false)
+        .identifiesControllerService(AzureCosmosDBConnectionService.class)
+        .build();
+
+
+    static final PropertyDescriptor URI = new PropertyDescriptor.Builder()
+        .name("azure-cosmos-db-uri")
+        .displayName("Azure Cosmos DB URI")
+        .description("Azure Cosmos DB URI, typically of the form: https://{databaseaccount}.documents.azure.com:443/. Note this host URL is for Azure Cosmos DB with CORE SQL API")
+        .required(false)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
```

Comment:
Would it make sense to use `URL_VALIDATOR` or `URI_VALIDATOR` here?

Reply:
fixed and moved to AzureCosmosDBUtils class.